gradio/components.py

```diff
@@ -4590,12 +4590,8 @@ def _postprocess_chat_messages(
                 "is_file": True,
             }
         elif isinstance(chat_message, str):
-            children = self.md.parseInline(chat_message)[0].children
-            if children and any("code" in child.tag for child in children):
-                return self.md.render(chat_message)
-            else:
-                chat_message = chat_message.replace("\n", "<br>")
-                return self.md.renderInline(chat_message)
+            chat_message = inspect.cleandoc(chat_message)
+            return self.md.render(chat_message)
```

Comment:
Just realized using `render` instead of `renderInline` returns results wrapped in `<p>` tags. But it's necessary to use `render` for fenced code blocks. So I can try to add a parser rule to remove `<p>` tags.